src/app.rs

```diff
@@ -1025,17 +1188,208 @@ impl eframe::App for ProfApp {
                 windows.push(Window::new(extra, index));
                 let window = windows.last_mut().unwrap();
                 cx.total_interval = cx.total_interval.union(window.config.interval);
-                cx.view_interval = cx.total_interval;
+                ProfApp::zoom(cx, cx.total_interval);
+            }
+
+            let visual = if cx.viewing_mode {
+                egui::Visuals::light()
+            } else {
+                egui::Visuals::dark()
+            };
+
+            // swap to dark mode
+            if let Some(toggled) = visual.light_dark_small_toggle_button(ui) {
+                ctx.set_visuals(toggled);
+                cx.viewing_mode = !cx.viewing_mode;
             }
 
-            if ui.button("Reset Zoom Level").clicked() {
-                cx.view_interval = cx.total_interval;
+            if ui.button("Reset Zoom Level").clicked() || ctx.input().key_pressed(egui::Key::Escape)
+            {
+                ProfApp::zoom(cx, cx.total_interval);
             }
 
             egui::Frame::group(ui.style()).show(ui, |ui| {
                 ui.set_width(ui.available_width());
                 ui.heading("Task Details");
                 ui.label("Click on a task to see it displayed here.");
+
+                let text_style = TextStyle::Body;
+                let row_height = ui.text_style_height(&text_style);
+
+                ui.separator();
+                ui.subheading("Search: ", cx);
+
+                let reply = ui.with_layout(egui::Layout::right_to_left(egui::Align::TOP), |ui| {
+                    if ui.button("X").clicked() {
```

Comment:
If we don't get around to finding a less hacky UI for this, let's add a TODO note in the README.